tools/scripts/Makefile

```diff
@@ -0,0 +1,265 @@
+CC ?= gcc
+
+# Xilinx platform handling
+ifeq (xilinx,$(strip $(PLATFORM)))
+
+CFLAGS = -Wall 								\
+	 -Wmissing-field-initializers					\
+	 -Wclobbered 							\
+	 -Wempty-body 							\
+	 -Wignored-qualifiers 						\
+	 -Wmissing-parameter-type					\
+	 -Wold-style-declaration					\
+	 -Woverride-init 						\
+	 -Wsign-compare							\
+	 -Wtype-limits							\
+	 -Wuninitialized						\
+	 -Wunused-but-set-parameter					\
+	 -Wno-unused-parameter						\
+	 -fdata-sections						\
+	 -ffunction-sections 						\
+	 -std=gnu90							\
+	 -O2 								\
+	 -g3								\
+	 -MMD 								\
+	 -MP								\
+	 -lm						
+	#-Werror
+
+LDFLAGS = -Wl,-T							\
+	  -Wl,$(BUILD_DIR)/app/src/lscript.ld -L $(LIBRARY_DIR)
+
+# Zynq
+ifeq (ps7_cortexa9_0,$(strip $(PROCESSOR_CORE)))
+
+CC = arm-none-eabi-gcc
+
+CFLAGS += -mcpu=cortex-a9 						\
+	  -mfpu=vfpv3 							\
+	  -mfloat-abi=hard
+
+LDFLAGS += -specs=$(BUILD_DIR)/app/src/Xilinx.spec 			\
+	   -mfpu=vfpv3							\
+ 	   -mfloat-abi=hard 						\
+	   -mcpu=cortex-a9						\
+	   -Wl,-build-id=none
+endif
+# Zynqmp 
+ifeq (psu_cortexa53_0,$(strip $(PROCESSOR_CORE)))
+
+CC = aarch64-none-elf-gcc
+
+endif
+# Microblaze
+ifeq (sys_mb,$(strip $(PROCESSOR_CORE)))
+
+CC = microblaze-xilinx-elf-gcc
+
+CFLAGS += -mcpu=cortex-a9 						\
+	  -DXILINX -DMICROBLAZE						\
+	  -DXILINX_PLATFORM						\
+	  -mlittle-endian						\
+	  -mxl-barrel-shift						\
+	  -mxl-pattern-compare 						\
+	  -mno-xl-soft-div						\
+	  -mcpu=v11.0							\
+	  -mno-xl-soft-mul						\
+	  -mxl-multiply-high
+
+LDFLAGS += -Xlinker --defsym=_HEAP_SIZE=0x100000 			\
+	   -Xlinker --defsym=_STACK_SIZE=0x2000 			\
+	   -mlittle-endian 						\
+	   -mxl-barrel-shift						\
+	   -mxl-pattern-compare						\
+	   -mno-xl-soft-div						\
+	   -mcpu=v11.0							\
+	   -mno-xl-soft-mul						\
+	   -mxl-multiply-high 						\
+	   -Wl,--no-relax 						\
+	   -Wl,--gc-sections 
+
+endif
+
+# Common xilinx libs
+LIBS =	-Wl,--start-group,-lxil,-lgcc,-lc,--end-group
+
+endif
+
+# Add the common include paths
+INC_PATHS = $(foreach x, $(INCS), $(addprefix -I ,$(dir $(x))))
+LIB_PATHS = $(foreach x, $(LIBRARY_DIR), $(addprefix -L ,$(x)))
+EXTRA_INCS =
+
+OBJS = $(SRCS:.c=.o)
+
+# Compile all the source files
+%.o : %.c
+	$(ECHO) [CC]	$(notdir $<) '\n'
+	$(MUTE)$(CC) $(CFLAGS) $(INC_PATHS) $(EXTRA_INCS)		\
+	-c $(addprefix $(BUILD_DIR)/app/src/,$(notdir $<))		\
+	-o $(addprefix $(OBJECTS_DIR)/,$(notdir $@))
+
+# Link the resulted object files
+$(EXEC): check-srcs $(subst LOCAL_PLATFORM,xilinx,$(OBJS))		\
+	$(BUILD_DIR)/app/src/lscript.ld
+	$(ECHO) [LD] $(shell ls $(OBJECTS_DIR)) '\n'
+	$(MUTE)$(CC) $(LDFLAGS) $(LIB_PATHS) 				\
+	$(OBJECTS_DIR)/*.o						\
+	$(LIBS) -o $(BUILD_DIR)/$(EXEC).elf
+
+# Copy the source files from the no-Os repo to the local project
+define copy_srcs
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(SRCS))		\
+	$(BUILD_DIR)/app/src/
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(INCS))		\
+	$(BUILD_DIR)/app/src/
```

Comment:
if breaking a new line, maybe indent the 2nd to make sure it's the same command;
maybe check this, in all the file